bench/workgen/workgen.cxx

```diff
@@ -775,30 +771,57 @@ int ThreadRunner::op_run(Operation *op) {
             op->kv_gen(this, false, compressibility, recno, _valuebuf);
             cursor->set_value(cursor, _valuebuf);
         }
-        switch (op->_optype) {
-        case Operation::OP_INSERT:
-            WT_ERR(cursor->insert(cursor));
-            break;
-        case Operation::OP_REMOVE:
-            WT_ERR_NOTFOUND_OK(cursor->remove(cursor));
-            break;
-        case Operation::OP_SEARCH:
-            ret = cursor->search(cursor);
-            break;
-        case Operation::OP_UPDATE:
-            WT_ERR_NOTFOUND_OK(cursor->update(cursor));
-            break;
-        default:
-            ASSERT(false);
-        }
-        if (ret != 0) {
-            ASSERT(ret == WT_NOTFOUND);
-            track = &_stats.not_found;
-            ret = 0;  // WT_NOTFOUND allowed.
-        }
-        cursor->reset(cursor);
-    } else
-        WT_ERR(op->_internal->run(this, _session));
+    }
+    // Retry on rollback until success.
+    while (retry_op) {
+	if (op->_transaction != NULL) {
+            if (_in_transaction)
+                THROW("nested transactions not supported");
+            _session->begin_transaction(_session,
+              op->_transaction->_begin_config.c_str());
+            _in_transaction = true;
+	}
+	if (op->is_table_op()) {
+	    bool notfound_ok = false;
+            switch (op->_optype) {
+            case Operation::OP_INSERT:
+                ret = cursor->insert(cursor);
+                break;
+            case Operation::OP_REMOVE:
+		notfound_ok = true;
+                ret = cursor->remove(cursor);
+                break;
+            case Operation::OP_SEARCH:
+                ret = cursor->search(cursor);
+                break;
+            case Operation::OP_UPDATE:
+		notfound_ok = true;
+                ret = cursor->update(cursor);
+                break;
+            default:
+                ASSERT(false);
+            }
+	    // Assume success and no retry unless ROLLBACK.
+            retry_op = false;
```

Comment:
I don't think this handles the `WT_NOTFOUND` case when `notfound_okay` is `false`.  For example, if there is a `cursor->search` that returns `WT_NOTFOUND`, an error is not reported, and it should be.

Reply:
@ddanderson can you look at the current version in the PR? I redid this and I believe it now matches what the original code does.